code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -531,17 +532,38 @@
 /datum/reagent/medicine/perfluorodecalin
 	name = "Perfluorodecalin"
 	id = "perfluorodecalin"
-	description = "Extremely rapidly restores oxygen deprivation, but inhibits speech. May also heal small amounts of bruising and burns."
+	description = "Restores oxygen deprivation while producing a lesser amount of toxic byproducts. Both scale with exposure to the drug and current amount of oxygen deprivation. Overdose causes toxic byproducts regardless of oxygen deprivation."
 	reagent_state = LIQUID
 	color = "#FF6464"
 	metabolization_rate = 0.25 * REAGENTS_METABOLISM
+	overdose_threshold = 35 // at least 2 full syringes +some, this stuff is nasty if left in for long
 
 /datum/reagent/medicine/perfluorodecalin/on_mob_life(mob/living/carbon/human/M)
-	M.adjustOxyLoss(-12*REM, 0)
-	M.adjustToxLoss(0.3*REM, 0)
+	var/oxycalc = 2.5*REM*current_cycle
+	oxycalc = min(oxycalc, M.oxyloss + PERF_BASE_DAMAGE) // this is so you always take at least PERF_BASE_DAMAGE tox even if you don't have oxygen loss to deter stacking (it is an emergency chem after all!)
+	if(oxycalc)
+		M.adjustOxyLoss(-oxycalc, 0)
+		M.adjustToxLoss(oxycalc/2.5, 0) //1*REM*current_cycle
+	if(prob(current_cycle) && M.losebreath)
+		M.losebreath--
 	..()
 	return TRUE
 
+/datum/reagent/medicine/perfluorodecalin/overdose_process(mob/living/M)
+	. = TRUE
+	var/oxycalc = 2.5*REM*current_cycle
+	M.adjustOxyLoss(-oxycalc, 0)
+	M.adjustToxLoss(oxycalc/2.5, 0)
+	//we then cancel out any healing done by on_mob_life since we want to simulate the same mechanism and NOT doubledip. losebreath is done regardless so *shrug*
+	oxycalc = min(oxycalc,M.oxyloss + PERF_BASE_DAMAGE)
+	if(oxycalc)
+		M.adjustOxyLoss(oxycalc, 0)
```

Comment:
wtf is this stuff moblife heals then od heals again and then damages ????

Reply:
I could remove the oxygen adjustments here since that wouldn't change but this is needed for tox.

Reply:
thats why i m asking, is the 2nd heal inside  the od an error?

Reply:
Conceptually, No. The goal was to make the toxin damage not be adjusted (lowered) based on the actual oxyloss healed during OD. To do this I would need to damage the amount healed in life in the OD process.

That said, I forgot if it's OD first then life because that would prob mess up the calc. I wonder if there's a variable that handles "Is this reagent in the OD phase" that I can check for...

Reply:
put a var/flag = 1 or some shit also this is really hard to understand cant you like do adjusttoxloss(5)  in the od ?

Reply:
```DM
/datum/reagent/medicine/perfluorodecalin/on_mob_life(mob/living/carbon/human/M)
    var/oxyhealing = 2.5*REM*current_cycle
    if(!overdosed)
        oxycalc = min(oxycalc,M.getOxyLoss()+PERF_BASE_DAMAGE) //if NOT overdosing, we lower our toxdamage to only the damage we actually healed with a minimum of 0.5. IE if we only heal 10 oxygen damage but we COULD have healed 20, we will only take toxdamage for the 10. We would take the toxdamage for the extra 10 if we were overdosing.
    M.adjustOxyLoss(-oxycalc, 0)
    M.adjustToxLoss(oxycalc/2.5, 0)
    //then the rest of the proc

/datum/reagent/medicine/perfluorodecalin/overdose_process(mob/living/M)
    metabolization_rate += 1   
    return ..()
```

that make more sense?